src/transaction/components/TransactionProvider.tsx

```diff
@@ -39,21 +40,49 @@ export function TransactionProvider({
   const [gasFee, setGasFee] = useState('');
   const [isToastVisible, setIsToastVisible] = useState(false);
 
-  const { status, writeContracts } = useWriteContracts({
+  const { status: statusWriteContracts, writeContracts } = useWriteContracts({
+    onError,
+    setErrorMessage,
+    setTransactionId,
+  });
+
+  const {
+    status: statusWriteContract,
+    writeContract,
+    data: writeContractTransactionHash,
+  } = useWriteContract({
     onError,
     setErrorMessage,
     setTransactionId,
   });
 
   const { transactionHash } = useCallsStatus({ onError, transactionId });
 
-  const handleSubmit = useCallback(() => {
+  const fallbackToWriteContract = useCallback(async () => {
+    for (const contract of contracts) {
+      try {
+        await writeContract(contract);
+      } catch (_err) {
+        setErrorMessage('genericErrorMessage');
+      }
+    }
+  }, [contracts, writeContract]);
+
+  const handleSubmit = useCallback(async () => {
     setErrorMessage('');
     setIsToastVisible(true);
-    writeContracts({
-      contracts,
-    });
-  }, [contracts, writeContracts]);
+    try {
+      const result = await writeContracts({
+        contracts,
+      });
+
+      if (result === undefined) {
+        await fallbackToWriteContract();
+      }
```

Comment:
This might seem strange. Note that when an EOA calls `writeContracts` the function does not fail. 

`result === undefined` is actually what is calling `fallbackToWriteContract`.

Reply:
I'm going to update that catch statement to error if it gets hit.